.azure-pipelines/client.yml

```diff
@@ -65,17 +65,23 @@ jobs:
         Linux:
           OSName: 'Linux'
           OSVmImage: 'ubuntu-16.04'
+          OSTestCommand: 'lsb_release'
         Windows:
           OSName: 'Windows'
           OSVmImage: 'vs2017-win2016'
+          OSTestCommand: 'ver'
         MacOs:
           OSName: 'MacOS'
           OSVmImage: 'macOS-10.13'
+          OSTestCommand: 'sw_vers'
 
     pool:
       vmImage: '$(OSVmImage)'
 
     steps:
+      - script: '$(OSTestCommand)'
```

Comment:
I believe this command will **print** the runtime OS information, however we want to **verify** the runtime OS.  Meaning, the build needs to fail with a clear error message if the runtime OS does not match what was requested.  I believe your previous PR was closer to this goal.